docs/reference/search/search-your-data/retrieve-selected-fields.asciidoc

```diff
@@ -167,6 +167,105 @@ no dedicated array type, and any field could contain multiple values. The
 a specific order. See the mapping documentation on <<array, arrays>> for more
 background.
 
+[discrete]
+[[search-fields-nested]]
+==== Handling of nested fields
+
+The `fields` response for <<nested,`nested` fields>> is slightly different from that
+of regular object fields. While leaf values inside regular `object` fields are
+returned as a flat list, values inside `nested` fields are grouped according
+to maintain the independence of each object inside the original nested array.
```

Comment:
I noticed a couple additional things:
* It could be clearer to explicitly map the subfields as we do in other places. That way users aren't wondering "what are these new `.keyword` fields"?
* Some early adopters expressed confusion over what the structure is for requests other than `*`. We could add a line like "The nested fields will be grouped by their nested paths, no matter what the pattern used to retrieve them" maybe even with an example?